jobs/async-upload/tests/integration/test_integration_async_upload.py

```diff
@@ -0,0 +1,378 @@
+):
+    """Test the complete async-upload job integration.
+    
+    This test:
+    1. Creates a RegisteredModel, ModelVersion, and placeholder ModelArtifact
+    2. Downloads and uploads an ONNX model to MinIO
+    3. Creates and applies a Kubernetes job using kustomize
+    4. Waits for job completion
+    5. Validates the final artifact state
+    """
+    
+    # Configuration
+    mr_host_url = os.environ.get("MR_HOST_URL", "http://localhost:8080")
+    container_image_uri = os.environ.get(
+        "CONTAINER_IMAGE_URI",
+        "ghcr.io/kubeflow/model-registry/job/async-upload:latest"
+    )
+    job_name = f"test-async-upload-job-{uuid.uuid4().hex[:8]}"
+    namespace = "default"
+    
+    # Step 1: Create RegisteredModel
+    print("Creating RegisteredModel...")
+    model_name = f"test-model-{uuid.uuid4().hex[:8]}"
+    rm = model_registry_client.register_model(
+        name=model_name,
+        uri="PLACEHOLDER",  # Will be updated by the job
+        version="v1.0.0",
+        model_format_name="onnx",
+        model_format_version="1.0",
+        description="Test model for async upload"
+    )
+    assert rm.id
+    print(f"  Created RegisteredModel with ID: {rm.id}")
+    
+    # Get the created model version and artifact
+    mv = model_registry_client.get_model_version(model_name, "v1.0.0")
+    assert mv and mv.id
+    print(f"  Created ModelVersion with ID: {mv.id}")
+    
+    ma = model_registry_client.get_model_artifact(model_name, "v1.0.0")
+    assert ma and ma.id
+    print(f"  Created ModelArtifact with ID: {ma.id}")
+    
+    # Verify initial state
+    assert ma.uri == "PLACEHOLDER"
+    assert ma.state == ArtifactState.UNKNOWN
+    
+    # Step 2: Download and upload ONNX model
+    print("Downloading ONNX model...")
+    with tempfile.TemporaryDirectory() as temp_dir:
+        model_file = Path(temp_dir) / "mnist-8.onnx"
+        
+        # Download the model
+        response = requests.get(
+            "https://github.com/onnx/models/raw/refs/heads/main/validated/vision/classification/mnist/model/mnist-8.onnx"
+        )
+        response.raise_for_status()
+        
+        with open(model_file, "wb") as f:
+            f.write(response.content)
+        
+        print("Uploading to MinIO...")
+        bucket = "default"
+        key = f"my-model/mnist-8.onnx"
+        upload_to_minio(str(model_file), bucket, key)
+        
+        # Step 3: Prepare job patches
+        print("Preparing job patches...")
+        base_job_path = Path(__file__).parent.parent.parent / "samples" / "sample_job_s3_to_oci.yaml"
+        
+        # Define patches to apply to the job
+        patches = {
+            "/metadata/name": job_name,
+            "/spec/template/spec/containers/0/image": container_image_uri,
+            "/spec/template/spec/containers/0/env/12/value": rm.id,
+            "/spec/template/spec/containers/0/env/13/value": mv.id,
+            "/spec/template/spec/containers/0/env/14/value": ma.id,
```

Comment:
can we patch for a json-path, possibly using some library, instead of hardcoding 12, 13, 14?
otherwise, please add a comment of the Env name for 12, 13, 14 for the next dev who will be looking into this :)